tools/ppls1/ppls1/fluids/shear_visco.py

```diff
@@ -6,30 +6,205 @@
+    'di': [2,10,12,2,1],
+    'li': [0,1,1,2,3]}
+    tab3['Nitrogen']=pd.DataFrame(data=d)
+
+    d={'i': [1,2,3,4,5,6],
+    'Ni': [12.19,13.99,0.005027,-18.93,-6.698,-3.827],
+    'ti': [0.42,0.0,0.95,0.5,0.9,0.8],
+    'di': [1,2,10,5,1,2],
+    'li': [0,0,0,2,4,4]}
+    tab3['Argon']=pd.DataFrame(data=d)
+
+    d={'i': [1,2,3,4,5],
+    'Ni': [17.67,0.4042,0.0001077,0.3510,-13.67],
+    'ti': [0.05,0.0,2.10,0.0,0.5],
+    'di': [1,5,12,8,1],
+    'li': [0,0,0,1,2]}
+    tab3['Oxygen']=pd.DataFrame(data=d)
+
+    d={'i': [1,2,3,4,5],
+    'Ni': [10.72,1.122,0.002019,-8.876,-0.02916],
+    'ti': [0.2,0.05,2.4,0.6,3.6],
+    'di': [1,4,9,1,8],
+    'li': [0,0,0,1,1]}
+    tab3['Air']=pd.DataFrame(data=d)
+
+    # tab4
+    tab4={'Nitrogen':None,'Argon':None,'Oxygen':None,'Air':None}
+    d={'i': [1,2,3,4,5,6,7,8,9],
+    'Ni': [1.511,2.117,-3.332,8.862,31.11,-73.13,20.03,-0.7096,0.2672],
+    'ti': [None,-1.0,-0.7,0.0,0.03,0.2,0.8,0.6,1.9],
+    'di': [None,None,None,1,2,3,4,8,10],
+    'li': [None,None,None,0,0,1,2,2,2]}
+    tab4['Nitrogen']=pd.DataFrame(data=d)
+
+    d={'i': [1,2,3,4,5,6,7,8,9,10],
+    'Ni': [0.8158,-0.4320,0.0,13.73,10.07,0.7375,-33.96,20.47,-2.274,-3.973],
+    'ti': [None,-0.77,-1.0,0.0,0.0,0.0,0.8,1.2,0.8,0.5],
+    'di': [None,None,None,1,2,4,5,6,9,1],
+    'li': [None,None,None,0,0,0,2,2,2,4]}
+    tab4['Argon']=pd.DataFrame(data=d)
+
+    d={'i': [1,2,3,4,5,6,7,8,9],
+    'Ni': [1.036,6.283,-4.262,15.31,8.898,-0.7336,6.728,-4.374,-0.4747],
+    'ti': [None,-0.9,-0.6,0.0,0.0,0.3,4.3,0.5,1.8],
+    'di': [None,None,None,1,3,4,5,7,10],
+    'li': [None,None,None,0,0,0,2,2,2]}
+    tab4['Oxygen']=pd.DataFrame(data=d)
+
+    d={'i': [1,2,3,4,5,6,7,8,9],
+    'Ni': [1.308,1.405,-1.036,8.743,14.76,-16.62,3.793,-6.142,-0.3778],
+    'ti': [None,-1.1,-0.3,0.1,0.0,0.5,2.7,0.3,1.3],
+    'di': [None,None,None,1,2,3,7,7,11],
+    'li': [None,None,None,0,0,2,2,2,2]}
+    tab4['Air']=pd.DataFrame(data=d)
+    
+    allFluids = ['Nitrogen','Argon','Oxygen','Air','LJTS','LJfull']
+    
+    if fluid not in allFluids:
+        print('Fluid not yet supported')
+        print('Possible fluids:')
+        print(allFluids)
+        return 0.0
+    
+    na=6.02214076e23
+    kb=1.380649e-23
+    
+    if fluid == 'LJTS':
+        fluid = 'Argon'
+        units = 'reduced'
+        sig=3.3916
+        eps=137.9
+        mass=39.948
+        tc=1.08
+        # rc=0.31
+    elif fluid == 'LJfull':
+        fluid = 'Argon'
+        units = 'reduced'
+        sig=3.3952
+        eps=116.79
+        mass=39.948
+        tc=1.32
+        # rc=0.31
+    else:
+        units = 'SI'
+        sig=1.0
+        eps=1.0
+        mass=1.0
+        tc=150.687
+    
+    if units == 'reduced':
+        T = T/tc*150.687
+        rho = rho/sig**3*1e30/na*1e-3
+        # refTime, refLambda: tref, lref
+        tref=sig*1e-10*np.sqrt(mass*1.660538e-27/(kb*eps))
+        eta_ref=1e3*kb/(sig*1e-10*tref) # [kg/m-s]  TODO
+    elif units == 'SI':
+        pass
```

Comment:
I guess this line raises the CodeQL error? What happens here? Shouldn't we return something? At least `None`?

Reply:
Thats a bit weird. This part of the could handles the input of the unit. If its neither "SI" nor "reduced", it should return an error or warning. Otherwise, it should go on.